yjit.c

```diff
@@ -300,6 +312,28 @@ rb_yjit_reserve_addr_space(uint32_t mem_size)
 #endif
 }
 
+// Allocate and initialize a bitmap for tracking the in-use cachelines of a CodeBlock.
+void *
+rb_yjit_alloc_code_bitmap(size_t size)
+{
+    return (void *)ZALLOC_N(char, size);
+}
+
+// Update the in-use cachelines of a CodeBlock. When all the bits for a memory page
+// become 0, we'll be able to free the page.
+//
+// This assumes that a single cacheline doesn't have code for different ISEQs. The
+// assumption should be maintained by calling cb.align_pos() at appropriate places.
+// TODO: We could add assertions that verify the assumption.
+void
+rb_yjit_update_code_bitmap(void *bitmap, size_t start_offset, size_t end_offset, bool used)
+{
+    // TODO: Update the corresponding bits of the given bitmap
+
+    // TODO: When all the bits for a memory page that overlaps the updated range
+    // become 0, deallocate the page, e.g. madvise with MADV_DONTNEED for Linux.
+}
```

Comment:
We may want to write that kind of logic in Rust just because we currently track the pages there (`virtualmem.rs`) ?

Reply:
Ah right. I assumed it's hard to modify the bitmap in Rust, but we could probably have this in Rust as long as we can read some specific byte from the bitmap, perform bitwise operations in Rust, and then write the byte back to the bitmap.